src/Stratis.Bitcoin/Consensus/Validators/BlockValidator.cs

```diff
@@ -165,6 +177,30 @@ public void StartPartialValidation(ChainedHeaderBlock chainedHeaderBlock, OnPart
             this.logger.LogTrace("(-)");
         }
 
+        /// <inheritdoc />
+        public PartialValidationResult ValidateSynchronously(Block block, ChainedHeader chainedHeader)
```

Comment:
can be async and renamed to validate instantly or smth like that. 
we just need to validate outside of async queue

Reply:
How about we name this just `ValidateAync`? and rename `StartPartialValidation` to `ValidateQueueAsync`?